src/lib_ccx/matroska.c

```diff
@@ -738,14 +738,35 @@ void parse_segment(struct matroska_ctx* mkv_ctx)
     }
 }
 
+char *filename_without_ext(char* filename) {
+    char *returnname;
+    char *lastdot;
+    if (filename == NULL)
+         return filename;
+    if ((returnname = malloc (strlen (filename) + 1)) == NULL)
```

Comment:
Yes, It is a big mistake. Would have caused errors among some system. Thanks for pointing